packages/grid/x-data-grid/src/hooks/features/focus/useGridFocus.ts

```diff
@@ -56,6 +56,15 @@ export const useGridFocus = (
         return;
       }
 
+      if (focusedCell) {
+        // There's a focused cell but another cell was clicked
+        // Publishes an event to notify that the focus was lost
+        apiRef.current.publishEvent(
+          'cellFocusOut',
```

Comment:
Adding the ref fixed the main problem but created another one. When the server responds, by default, the grid focus the cell that is below the cell being edited, this means that the new cell being edited (that one that entered edit mode while waiting `processRowUpdate`) lost focus. When a cell loses focus it should commit the value but it doesn't do this because `cellFocusOut` is only published when there's a click, not when the focus is changed programmatically, so I'm also publishing the event here now. 

This may create an unwanted behavior because it commits the next cell when the server responds because of the first. However, this can be solved in two ways:
- 1st option: Use row editing mode
- 2nd option: Cancel the default behavior in `onCellEditStop` and manually call `apiRef.current.stopCellEditMode` without passing which cell to focus after.